pkg/clusterapi/aws/actuator_test.go

```diff
@@ -0,0 +1,351 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+
+	clustopv1 "github.com/openshift/cluster-operator/pkg/apis/clusteroperator/v1alpha1"
+	"github.com/openshift/cluster-operator/pkg/controller"
+
+	capicommon "sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
+	capiv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	capiclientfake "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/fake"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	clientgofake "k8s.io/client-go/kubernetes/fake"
+)
+
+const (
+	testNamespace             = "test-namespace"
+	testClusterDeploymentName = "test-cluster-deployment"
+	testClusterDeploymentUUID = types.UID("test-cluster-deployment-uuid")
+	testClusterVerName        = "v3-10"
+	testClusterVerNS          = "cluster-operator"
+	testClusterVerUID         = types.UID("test-cluster-version")
+	testRegion                = "us-east-1"
+	testImage                 = "testAMI"
+	testVPCID                 = "testVPCID"
+	testSubnetID              = "testSubnetID"
+	testAZ                    = "us-east-1c"
+)
+
+func TestBuildDescribeSecurityGroupsInput(t *testing.T) {
+	cases := []struct {
+		name               string
+		isMaster           bool
+		isInfra            bool
+		expectedGroupNames []string
+	}{
+		{
+			name:               "master",
+			isMaster:           true,
+			expectedGroupNames: []string{"cn", "cn_master", "cn_master_k8s"},
+		},
+		{
+			name:               "master and infra",
+			isMaster:           true,
+			isInfra:            true,
+			expectedGroupNames: []string{"cn", "cn_master", "cn_master_k8s", "cn_infra", "cn_infra_k8s"},
+		},
+		{
+			name:               "infra",
+			isInfra:            true,
+			expectedGroupNames: []string{"cn", "cn_infra", "cn_infra_k8s"},
+		},
+		{
+			name:               "compute",
+			expectedGroupNames: []string{"cn", "cn_compute", "cn_compute_k8s"},
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			input := buildDescribeSecurityGroupsInput("", "cn", tc.isMaster, tc.isInfra)
+			groupNamesFilter, err := findFilter(input.Filters, "group-name")
+			if assert.NoError(t, err) {
+				assert.Equal(t, len(tc.expectedGroupNames), len(groupNamesFilter.Values))
+				for i, sgn := range groupNamesFilter.Values {
+					assert.Equal(t, tc.expectedGroupNames[i], *sgn)
```

Comment:
This is still requiring that the groups are in a particular order. I recommend creating a `[]string` from `groupNamesFilter.Values` and then simply calling `assert.ElementsMatch`.